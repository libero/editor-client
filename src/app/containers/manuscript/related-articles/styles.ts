import { makeStyles } from '@material-ui/core/styles';

export const useRelatedArticleStyles = makeStyles((theme) => ({
  addButton: {
    marginLeft: theme.spacing(1)
  },
  list: {
    marginTop: 0,
    marginLeft: theme.spacing(3),
    minHeight: theme.spacing(2.75),
    listStyle: 'disc',
    '& li + li': {
      marginTop: theme.spacing(1)
    }
  },
  listItem: {
    display: 'flex',
    alignItems: 'flex-end',
    lineHeight: '24px',
    paddingLeft: theme.spacing(1),
    marginBottom: theme.spacing(1),
    '&:last-of-type': {
      marginBottom: 0
    }
  },
  info: {
    flex: 1
  },
  editButton: {
    flexGrow: 0,
    padding: 6,
    marginLeft: theme.spacing(2)
  }
}));
