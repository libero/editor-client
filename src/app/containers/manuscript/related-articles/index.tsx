import React, { useCallback } from 'react';
import { useSelector } from 'react-redux';
import { IconButton } from '@material-ui/core';
import EditIcon from '@material-ui/icons/Edit';

import { SectionContainer } from 'app/components/section-container';
import { getRelatedArticles } from 'app/selectors/manuscript.selectors';
import { useRelatedArticleStyles } from 'app/containers/manuscript/related-articles/styles';
import { RelatedArticle } from 'app/models/related-article';
import { ActionButton } from 'app/components/action-button';

const getUrl = (href: string, type: string) => {
  return (
    {
      doi: `https://doi.org/${href}`
    }[type] || href
  );
};

export const RelatedArticles: React.FC = () => {
  const relatedArticles = useSelector(getRelatedArticles);
  const classes = useRelatedArticleStyles();

  const handleEditRelatedArticle = useCallback((relatedArticle: RelatedArticle) => {}, []);
  const handleAddRelatedArticle = useCallback(() => {}, []);

  return (
    <section>
      <SectionContainer label="Related articles">
        <ul className={classes.list}>
          {relatedArticles.map((relatedArticle) => (
            <li>
              <div className={classes.listItem}>
                <div className={classes.info}>
                  {relatedArticle.linkType}:{' '}
                  <a
                    rel="noopener noreferrer"
                    target="_blank"
                    href={getUrl(relatedArticle.href, relatedArticle.linkType)}
                  >
                    {getUrl(relatedArticle.href, relatedArticle.linkType)}
                  </a>{' '}
                  type: {relatedArticle.articleType}
                </div>
                <IconButton
                  classes={{ root: classes.editButton }}
                  onClick={handleEditRelatedArticle.bind(null, relatedArticle)}
                >
                  <EditIcon fontSize="small" />
                </IconButton>
              </div>
            </li>
          ))}
        </ul>
      </SectionContainer>
      <ActionButton variant="addEntity" title="Article" onClick={handleAddRelatedArticle} />
    </section>
  );
};
