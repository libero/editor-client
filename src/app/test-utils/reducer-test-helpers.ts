import { EditorState } from 'prosemirror-state';
import { Schema } from 'prosemirror-model';

import { Manuscript } from 'app/models/manuscript';
import { getInitialHistory, getLoadableStateSuccess } from 'app/utils/state.utils';
import { ManuscriptHistoryState } from 'app/store';

export function givenState(overrides: Partial<Manuscript>): ManuscriptHistoryState {
  return getLoadableStateSuccess(
    getInitialHistory({
      title: createDummyEditorState(),
      abstract: createDummyEditorState(),
      authors: [],
      affiliations: [],
      keywordGroups: {},
      articleInfo: {
        dtd: '',
        articleType: '',
        articleDOI: ''
      },
      references: [],
      ...overrides
    })
  );
}

export function createDummyEditorState(): EditorState {
  return EditorState.create({
    schema: new Schema({
      nodes: {
        doc: { content: 'inline*' },
        text: { inline: true, group: 'inline' }
      }
    }),
    plugins: []
  });
}
