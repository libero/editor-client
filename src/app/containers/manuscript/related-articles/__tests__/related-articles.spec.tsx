import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import { create } from 'react-test-renderer';
import { givenState } from 'app/test-utils/reducer-test-helpers';
import { RelatedArticles } from 'app/containers/manuscript/related-articles/index';

jest.mock('@material-ui/core', () => ({
  Button: ({ label }) => <div data-cmp="Button">{label}</div>,
  IconButton: () => <div data-cmp="IconButton"></div>
}));

const RELATED_ARTICLES = [
  {
    linkType: 'doi',
    articleType: 'commentary-article',
    href: '10.7554/eLife.42697'
  },
  {
    linkType: 'doi',
    articleType: 'commentary-article',
    href: '10.7554/eLife.00067'
  }
];

describe('Related articles info', () => {
  const mockStore = configureMockStore([]);

  it('renders related articles list', () => {
    const mockState = givenState({});
    const store = mockStore({
      manuscript: mockState
    });
    const wrapper = create(
      <Provider store={store}>
        <RelatedArticles />
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
