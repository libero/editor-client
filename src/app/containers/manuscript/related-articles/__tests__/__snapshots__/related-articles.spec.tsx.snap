// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Related articles info renders related articles list 1`] = `
<section>
  <div
    className="makeStyles-root-6 makeStyles-plainVariant-8"
  >
    <label
      className="makeStyles-label-9"
    >
      Related articles
    </label>
    <ul
      className="makeStyles-list-2"
    />
  </div>
  <div
    data-cmp="Button"
  />
</section>
`;
